realm-annotations-processor/src/test/resources/io/realm/NullTypesRealmProxy.java

```diff
@@ -0,0 +1,1257 @@
+        fieldNames.add("fieldStringNull");
+        fieldNames.add("fieldBooleanNotNull");
+        fieldNames.add("fieldBooleanNull");
+        fieldNames.add("fieldBytesNotNull");
+        fieldNames.add("fieldBytesNull");
+        fieldNames.add("fieldByteNotNull");
+        fieldNames.add("fieldByteNull");
+        fieldNames.add("fieldShortNotNull");
+        fieldNames.add("fieldShortNull");
+        fieldNames.add("fieldIntegerNotNull");
+        fieldNames.add("fieldIntegerNull");
+        fieldNames.add("fieldLongNotNull");
+        fieldNames.add("fieldLongNull");
+        fieldNames.add("fieldFloatNotNull");
+        fieldNames.add("fieldFloatNull");
+        fieldNames.add("fieldDoubleNotNull");
+        fieldNames.add("fieldDoubleNull");
+        fieldNames.add("fieldDateNotNull");
+        fieldNames.add("fieldDateNull");
+        fieldNames.add("fieldObjectNull");
+        FIELD_NAMES = Collections.unmodifiableList(fieldNames);
+    }
+
+    @Override
+    public String getFieldStringNotNull() {
+        realm.checkIfValid();
+        return (java.lang.String) row.getString(INDEX_FIELDSTRINGNOTNULL);
+    }
+
+    @Override
+    public void setFieldStringNotNull(String value) {
+        realm.checkIfValid();
+        if (value == null) {
+            throw new IllegalArgumentException("Trying to set non-nullable field fieldStringNotNull to null.");
+        }
+        row.setString(INDEX_FIELDSTRINGNOTNULL, (String) value);
+    }
+
+    @Override
+    public String getFieldStringNull() {
+        realm.checkIfValid();
+        return (java.lang.String) row.getString(INDEX_FIELDSTRINGNULL);
+    }
+
+    @Override
+    public void setFieldStringNull(String value) {
+        realm.checkIfValid();
+        if (value == null) {
+            row.setNull(INDEX_FIELDSTRINGNULL);
+            return;
+        }
+        row.setString(INDEX_FIELDSTRINGNULL, (String) value);
+    }
+
+    @Override
+    public Boolean getFieldBooleanNotNull() {
+        realm.checkIfValid();
+        return (boolean) row.getBoolean(INDEX_FIELDBOOLEANNOTNULL);
+    }
+
+    @Override
+    public void setFieldBooleanNotNull(Boolean value) {
+        realm.checkIfValid();
+        if (value == null) {
+            throw new IllegalArgumentException("Trying to set non-nullable field fieldBooleanNotNull to null.");
+        }
+        row.setBoolean(INDEX_FIELDBOOLEANNOTNULL, (Boolean) value);
+    }
+
+    @Override
+    public Boolean getFieldBooleanNull() {
+        realm.checkIfValid();
+        if (row.isNull(INDEX_FIELDBOOLEANNULL)) {
+            return null;
+        }
+        return (boolean) row.getBoolean(INDEX_FIELDBOOLEANNULL);
+    }
+
+    @Override
+    public void setFieldBooleanNull(Boolean value) {
+        realm.checkIfValid();
+        if (value == null) {
+            row.setNull(INDEX_FIELDBOOLEANNULL);
+            return;
+        }
+        row.setBoolean(INDEX_FIELDBOOLEANNULL, (Boolean) value);
```

Comment:
For no reason... I can remove it in another PR. this PR is big enough and the cast is there before.